WordPress/Classes/ViewRelated/Post/WPStyleGuide+Posts.swift

```diff
@@ -0,0 +1,140 @@
+        label.textColor = darkGrey()
+    }
+
+    class func applyPostDateStyle(_ label: UILabel) {
+        configureLabelForRegularFontStyle(label)
+        label.textColor = grey()
+    }
+
+    class func applyPostButtonStyle(_ button: UIButton) {
+        configureLabelForRegularFontStyle(button.titleLabel)
+        button.setTitleColor(grey(), for: .normal)
+    }
+
+    class func applyPostProgressViewStyle(_ progressView: UIProgressView) {
+        progressView.trackTintColor = greyLighten20()
+        progressView.progressTintColor = mediumBlue()
+        progressView.tintColor = mediumBlue()
+    }
+
+    class func applyRestorePostLabelStyle(_ label: UILabel) {
+        configureLabelForRegularFontStyle(label)
+        label.textColor = grey()
+    }
+
+    class func applyRestorePostButtonStyle(_ button: UIButton) {
+        configureLabelForRegularFontStyle(button.titleLabel)
+        button.setTitleColor(wordPressBlue(), for: .normal)
+        button.setTitleColor(darkBlue(), for: .highlighted)
+    }
+
+    class func applyBorderStyle(_ view: UIView) {
+        view.heightAnchor.constraint(equalToConstant: 1.0 / UIScreen.main.scale).isActive = true
+        view.backgroundColor = postCardBorderColor()
+    }
+
+    class func applyActionBarButtonStyle(_ button: UIButton) {
+        button.flipInsetsForRightToLeftLayoutDirection()
+        button.setImage(button.imageView?.image?.imageWithTintColor(grey()), for: .normal)
+        button.setTitleColor(grey(), for: .normal)
+        button.setTitleColor(darkGrey(), for: .highlighted)
+        button.setTitleColor(darkGrey(), for: .selected)
+    }
+
+    class func applySelectedBackgroundView(_ selectedBackgroundView: UIView, topMargin: CGFloat) {
+        let marginMask = UIView()
+        selectedBackgroundView.addSubview(marginMask)
+        marginMask.translatesAutoresizingMaskIntoConstraints = false
+        marginMask.leadingAnchor.constraint(equalTo: selectedBackgroundView.leadingAnchor).isActive = true
+        marginMask.topAnchor.constraint(equalTo: selectedBackgroundView.topAnchor).isActive = true
+        marginMask.trailingAnchor.constraint(equalTo: selectedBackgroundView.trailingAnchor).isActive = true
+        marginMask.heightAnchor.constraint(equalToConstant: topMargin).isActive = true
+        marginMask.backgroundColor = greyLighten30()
+    }
+
+    // MARK: - Attributed String Attributes
+
+    class var postCardTitleAttributes: [NSAttributedString.Key: Any] {
+        let font = notoBoldFontForTextStyle(.headline)
+        let lineHeight = 1.4 * font.pointSize
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.minimumLineHeight = lineHeight
+        paragraphStyle.maximumLineHeight = lineHeight
+        return [.paragraphStyle: paragraphStyle, .font: font]
+    }
+
+    class var postCardSnippetAttributes: [NSAttributedString.Key: Any] {
+        let textStyle: UIFont.TextStyle = UIDevice.isPad() ? .callout : .subheadline
```

Comment:
Do we want to do a device check here? Is the font intended to be the same on iPad in slide over mode as well?

Reply:
This is a really good question!

What I did here was to translate the previous existent logic. What are your thoughts on this, @SylvesterWilmott?

JFYI, the Reader uses `.subheadline` regardless of the device.

Reply:
I'd like to heard what Sylvester thinks but I know he's at a meetup this week. I think we can leave this for now and hopefully come back to it in the future.